src/context/AuthContext.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useState } from "react";
+import jwt_decode from "jwt-decode";
+import { APIUrl } from "../config/config";
+
+export const AuthContext = createContext({});
+
+export default function AuthContextProvider({ children }) {
```

Comment:
I am missing what this AuthContext does on startup

I think it should fetch your userdata when the app starts up and maybe check if your token is still valid